lib/cppcheck.cpp

```diff
@@ -1418,8 +1418,7 @@ void CppCheck::executeAddons(const std::vector<std::string>& files, const std::s
     std::string fileList;
 
     if (files.size() >= 2 || endsWith(files[0], ".ctu-info")) {
-        fileList = Path::getPathFromFilename(files[0]) + FILELIST + std::to_string(mSettings.pid);
```

Comment:
Removing the unique identifier (pid) from the filename re-introduces the issue this was fixing - parallel executions writing the same file. This is exposed by running the Python tests with `python-xdist` - see #6696.

Reply:
oh no!! :-(

Reply:
I think I am seeing the same bug, but not totally sure.

This repo helps reproduce a concurrency problem on my Linux machine
https://github.com/andymacg/cppcheck-addon-concurrency-bug

Reply:
A quick fix is simply adding back the pid to the filename. Will take a look later - I totally forgot about it.

Reply:
> A quick fix is simply adding back the pid to the filename. Will take a look later - I totally forgot about it.

I have done that with #7082. Feel free to review it..